modules/mongodb/src/test/java/org/testcontainers/containers/AtlasLocalDataAccess.java

```diff
@@ -0,0 +1,171 @@
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.ListSearchIndexesIterable;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import lombok.Getter;
+import lombok.Setter;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.PojoCodecProvider;
+import org.bson.conversions.Bson;
+import org.bson.json.JsonWriterSettings;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+import static com.mongodb.client.model.Aggregates.search;
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.search.SearchOperator.of;
+import static com.mongodb.client.model.search.SearchOperator.text;
+import static com.mongodb.client.model.search.SearchOptions.searchOptions;
+import static com.mongodb.client.model.search.SearchPath.fieldPath;
+import static java.lang.Thread.sleep;
+import static java.util.Collections.singletonList;
+import static java.util.Objects.requireNonNull;
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class AtlasLocalDataAccess implements AutoCloseable {
+
+    private static final Logger log = getLogger(AtlasLocalDataAccess.class);
+    private final MongoClient mongoClient;
+    private final MongoDatabase testDB;
+    private final MongoCollection<TestData> testCollection;
+    private final String collectionName;
+
+    public AtlasLocalDataAccess(String connectionString, String databaseName, String collectionName) {
+        this.collectionName = collectionName;
+        log.info("DataAccess connecting to {}", connectionString);
+
+        CodecRegistry pojoCodecRegistry = fromProviders(PojoCodecProvider.builder().automatic(true).build());
+        CodecRegistry codecRegistry = fromRegistries(MongoClientSettings.getDefaultCodecRegistry(), pojoCodecRegistry);
+        MongoClientSettings clientSettings = MongoClientSettings
+            .builder()
+            .applyConnectionString(new ConnectionString(connectionString))
+            .codecRegistry(codecRegistry)
+            .build();
+        mongoClient = MongoClients.create(clientSettings);
+        testDB = mongoClient.getDatabase(databaseName);
+        testCollection = testDB.getCollection(collectionName, TestData.class);
+    }
+
+    @Override
+    public void close() {
+        mongoClient.close();
+    }
+
+    // initAtlasSearchIndex {
+    public void initAtlasSearchIndex() throws URISyntaxException, IOException, InterruptedException {
```

Comment:
Currently, databases provide some default values for username, password and databases. What if we do the same for MongoDB Atlas implementation providing default values for database, collection name that can be customized? Also, a `withIndex(Transferable json)` copies the mapping json file from classpath to the container and execute the right commands in the container. No need to do so right away but just thinking out loud.

Reply:
would be also nice to see how we can configure credentials support. I see langchain4j using it https://github.com/langchain4j/langchain4j/blob/main/langchain4j-mongodb-atlas/src/test/java/dev/langchain4j/store/embedding/mongodb/MongoDbEmbeddingStoreLocalIT.java#L55-L61

Reply:
I've spoken to MongoDB and requested that they add the MongoDB shell mongosh, as well as the database tools mongoimport, mongoexport, mongodump... which would allow us to create these nice extra features. If they add these to the image, I will add these features as a further PR.

Reply:
ok, nice! any public place where we can follow the request?

I'm very interested on this because would like to improve the time to execute test in langchain4j and spring-ai integrations.

Reply:
I shared this request with the product team directly, but it's a great idea to have a ticket - I created one and will share with the product team too:
https://feedback.mongodb.com/forums/924868-atlas-search/suggestions/48857015-include-mongodb-shell-mongosh-and-mongodb-tools